### PR TITLE
Add basic implementation of AUTOSAR SecOC

### PR DESCRIPTION
**Pull request by polybassa**
This PR adds a very basic implementation of AUTOSAR SecOC packets.

Basic therefore, because the actual implementation of the message authentication code calculation is left to the user, since there is no standard across OEMs.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -59,6 +59,7 @@ all = [
     "matplotlib",
 ]
 doc = [
+    "cryptography>=2.0",
```
Comment (gpotter2) [inline pyproject.toml:62]: This doesn't look right

File: scapy/contrib/automotive/autosar/pdu.py [hunk 2]
```diff
@@ -26,7 +26,7 @@ class PDU(Packet):
     name = 'PDU'
     fields_desc = [
         XIntField('pdu_id', 0),
-        IntField('pdu_payload_len', 0)]
+        LenField('pdu_payload_len', 0, fmt="I")]
```
Comment (gpotter2) [inline scapy/contrib/automotive/autosar/pdu.py:29]: You probably want to replace 0 with None for this to do something (?)

File: pyproject.toml (hunk 1 above)
Comment (gpotter2) [reply pyproject.toml:62]: This is pretty normal. `cryptography` isn't a hard dependency (Scapy has None), so you should be able to load any of the Scapy's module/layers/contrib without having them crashing.

See https://github.com/secdev/scapy/blob/4bc73eda1829627381e8f61cd1ab4e2d7bebf524/scapy/config.py#L1164 and everywhere it's used. Usually the import is inside a try:catch, then we have that decorator for each function it's used in.

**Comment by polybassa** [comment]
@gpotter2 Ready for review.

**Review by gpotter2 — APPROVED** [review]

**Pull request merged**